runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3054,7 +3054,14 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
 
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
```

Comment:
You can put a comment that while we update parameters that represent the machine/container characteristics (like memory available), we don't change the original heap geometry (Xmx etc) from snapshot run, Instead, we just force that heap does not grow beyond a certain thereshold calculated based on the new memory availbility, using SoftMx functionality.

Reply:
Can we add another high lever description comment regarding why we softmx is used to adjust heap size at restore path, reflecting the fact that we are re-using existed softmx logic to adjust default maximum heap size in restore path in smaller memory environments temporary. The real solution should be adjustment of real heap geometry.